processing/src/main/java/io/druid/segment/Metadata.java

```diff
@@ -137,19 +164,25 @@ public boolean equals(Object o)
 
     Metadata metadata = (Metadata) o;
 
-    if (!container.equals(metadata.container)) {
+    if (container != null ? !container.equals(metadata.container) : metadata.container != null) {
       return false;
     }
     // Probably incorrect - comparing Object[] arrays with Arrays.equals
```

Comment:
Why is it incorrect - `Arrays.equals` will use `equals` method of `AggregatorFactory` ?